src/event.rs

```diff
@@ -58,49 +57,36 @@ pub enum Event {
 
 impl fmt::Display for Event {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        match *self {
+        match self {
             Event::BootstrapFailure => write!(f, "Event::BootstrapFailure"),
-            Event::BootstrappedTo { ref node } => {
+            Event::BootstrappedTo { node } => {
                 write!(f, "Event::BootstrappedTo {{ node: {} }}", node)
             }
-            Event::ConnectionFailure {
-                ref peer_addr,
-                ref err,
-            } => write!(
+            Event::ConnectionFailure { peer, err } => write!(
                 f,
-                "Event::ConnectionFailure {{ peer_addr: {}, err: {} }}",
-                peer_addr, err
+                "Event::ConnectionFailure {{ peer: {}, err: {} }}",
+                peer.peer_addr(),
```

Comment:
Is there a reason to display only the address specifically? Given that `Peer` implements `Display` maybe it will be more informative to display it entirely, with the metainformation like `Peer::Node` or `Peer::Client`.

Reply:
I guess just to preserve the previous behavior. But probably makes sense to do what you say.